pkg/collector/corechecks/net/ntp_test.go

```diff
@@ -336,6 +337,11 @@ hosts:
 }
 
 func TestDefaultHostConfig(t *testing.T) {
+	// Reset any cached cloud-provider information.  All clouds default to
+	// disabled, so without any cached data the Configure call below should
+	// fall back to the default configuration.
+	cloudproviders.ResetCaches()
```

Comment:
Did you consider mocking https://github.com/DataDog/datadog-agent/blob/7.23.0/pkg/collector/corechecks/net/ntp.go#L79?

Reply:
Yes -- that would be more unit-testy, but it's also a more invasive change that affects production code as well, so this seemed like the better choice.  WDYT?

Reply:
OK, that was pretty easy -- #10116